File: apps/auth-server/src/main/java/com/example/chat/auth/server/api/dto/request/SignupRequest.java
package com.example.chat.auth.server.api.dto.request;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;

public record SignupRequest(
    @NotBlank @Email String email,
    @NotBlank String password,
    @NotBlank String nickname
) {
    public SignupRequest {
        
    }
}

File: apps/auth-server/src/main/kotlin/com/example/chat/auth/server/api/dto/request/SignupRequest.kt
package com.example.chat.auth.server.api.dto.request

import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank

data class SignupRequest(
	@field:NotBlank
	@field:Email
	val email: String,

	@field:NotBlank
	val password: String,

	@field:NotBlank
	val nickname: String
)
